shade-test/jar-file/src/test/java/org/testcontainers/PublicAPITest.java

```diff
@@ -0,0 +1,120 @@
+package org.testcontainers;
+
+import lombok.RequiredArgsConstructor;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldNode;
+import org.objectweb.asm.tree.MethodNode;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+@RequiredArgsConstructor
+public class PublicAPITest extends AbstractJarFileTest {
```

Comment:
I do like the improved readability of @asafm's suggestion, but also am quite comfortable with `@Parameterized` and the reporting benefits it yields.

Perhaps one small step towards better readability we could take would be to do something with the `SimpleFileVisitor` anon class that's sitting in the middle of the test. It feels like it causes a bit of a cognitive load to context switch around it. Could readability be improved by something simple like extracting it to an obviously named top-level class/static utility method?